packages/dashboards/addon/routes/dashboards/dashboard/view.js

```diff
@@ -13,18 +15,146 @@ export default Route.extend({
 
   /**
-   * Makes an ajax request to retrieve relevant widgets in the dashboard
+   * @property {Service} compression
+   */
+  compression: service(),
+
+  /**
+   * @property {Service} metadataService
+   */
+  metadataService: service('bard-metadata'),
+
+  /**
+   * @property {Service} store
+   */
+  store: service(),
+
+  /**
+   * @property {Service} naviNotifications
+   */
+  naviNotifications: service(),
+
+  /**
+   * @private
+   * @property {Object} _widgetDataCache - stores most recent widget data
+   */
+  _widgetDataCache: null,
+
+  /**
+   * @override
+   * @property {Object} queryParams - any time the filters query params changes, rerun the model hook
+   */
+  queryParams: {
+    filters: {
+      refreshModel: true
+    }
+  },
+
+  /**
+   * Adds filters from query params to the dashboard then
+   * makes an ajax request to retrieve relevant widgets in the dashboard
    *
-   * @method model
    * @override
+   * @method model
+   * @param {Object} transition
    */
-  model() {
-    let dashboard = this.modelFor('dashboards.dashboard');
+  async model(_, transition) {
+    const dashboard = this.modelFor('dashboards.dashboard');
+    const filterQueryParams = get(transition, 'queryParams.filters');
+    const cachedWidgetData = this.get('_widgetDataCache');
 
-    return this.get('dashboardData')
-      .fetchDataForDashboard(dashboard)
-      .then(dataForWidget => {
-        return { dashboard, dataForWidget };
+    //Record filters before and after setting from query params
+    const originalFilters = dashboard.get('filters').serialize();
+    try {
+      await this._addFiltersFromQueryParams(dashboard, filterQueryParams);
+    } catch (e) {
+      get(this, 'naviNotifications').add({
+        message: `Error decoding filter query params. Using default dashboard filters`,
+        type: 'danger',
+        timeout: 'medium'
       });
+      throw e;
+    }
+    const newFilters = dashboard.get('filters').serialize();
+
+    const originalValuelessFilters = originalFilters.filter(fil => fil.values.length === 0);
+    const newValuelessFilters = newFilters.filter(fil => fil.values.length === 0);
+
+    const wasEmptyFilterAdded =
+      newValuelessFilters.length === originalValuelessFilters.length + 1 &&
+      newFilters.length === originalFilters.length + 1;
+    const wasEmptyFilterRemoved =
+      newValuelessFilters.length === originalValuelessFilters.length - 1 &&
+      newFilters.length === originalFilters.length - 1;
+
+    /**
+     * If we just added a new filter (no values yet) or just removed a filter with no values,
+     * return the last set of widget data because the requests will not have changed because
+     * empty filters are pruned from the request
+     */
+    if (cachedWidgetData && (wasEmptyFilterAdded || wasEmptyFilterRemoved)) {
+      return { dashboard, dataForWidget: cachedWidgetData };
+    }
+    const dataForWidget = await this.get('dashboardData').fetchDataForDashboard(dashboard);
```

Comment:
Caching down a layer to only re-fetch widget request that changed?

Reply:
Will be done in a later PR as an optimization